### ISSUE TITLE
Unmarshal works incorrectly in some cases

### ISSUE THREAD

**Issue by aegoroff** [pelletier/go-toml#249]
**Describe the bug**
Maps unmarshalling works incorrectly in some cases

**To Reproduce**
Here is the code:
```
type tomlConfig struct {
	Title       string
	Sources     map[string]source
	Definitions map[string]definition
}

type source struct {
	Source string
}

type definition struct {
	SourceLink string
	Source     string
	Target     string
}

doc := []byte(`# test config

title = "test"

[sources]
 [sources.source1]
  source = 's'

[definitions]

  [definitions.def1]
  sourceLink = "source1"
  target = 't'

  [definitions.def2]
  source = 's1'
  target = 't1'`)

config := Config{}
toml.Unmarshal(doc, &config)
fmt.Printf("%v\n", config)
```

**Expected behavior**
I expect output to be **{test map[source1:{s}] map[def2:{ s1 t1} def1:{ source1 t}]}** but i got **{test map[source1:{s}] map[def2:{ s1 t1} def1:{ t}]}** i.e. element def1 doesn't contain correct sourceLink value.

**Versions**
 - go-toml: latest repository version
 - go: 1.11
 - operating system: e.g. Windows

**Additional context**
Unmarshalling using "github.com/BurntSushi/toml" works correctly

**Comment by tom5760** [comment]
I think this is because `go-toml` doesn't handle the `camelCased` `sourceLink` value.  I just ran into something similar.  If you change `sourceLink` in your TOML file to either `sourcelink` or `SourceLink`, I bet it will work.

In `Decoder.valueFromTree` in `marshal.go`, there's a `keysToTry` value that tries to look up keys in the tree based on struct fields.  Right now it is:

```go
keysToTry := []string{baseKey, strings.ToLower(baseKey), strings.ToTitle(baseKey)}
```

Maybe something like the following would work?

```go
keysToTry := []string{
    baseKey,
    strings.ToLower(baseKey),
    strings.ToTitle(baseKey),
    strings.ToLower(baseKey[0]) + baseKey[1:],
}
```

**Comment by aegoroff** [comment]
You are right! it setting sourceLink key to sourceLink = "source1" helped me. I decided not to use keysToTry but if so i suppose it will help.
